nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/TestConvertRecord.java

```diff
@@ -62,6 +62,44 @@ public void testSuccessfulConversion() throws InitializationException {
         out.assertContentEquals("header\nJohn Doe,48\nJane Doe,47\nJimmy Doe,14\n");
     }
 
+    @Test
+    public void testDropEmpty() throws InitializationException {
```

Comment:
I think it's possible to have a ConvertRecord scenario where there are valid records coming in, but the conversion produces no records (no fields match the schema, e.g.), @markap14 do you know if that's possible or if an error would occur? If it is possible, it would be nice to have a unit test that covers that.

Reply:
This is probably possible with the write record reader configuration. I tried building a test using Json Row Record Reader, but even if none of the elements in your schema match the schema of the source record it just outputs a NULL for you...

```Successfully converted 2 records for FlowFile[0,627269493239800.mockFlowFile,119B]
valuated JSONPath Expression com.jayway.jsonpath.JsonPath@5b8d184c but the path was not found; will use a null value
Evaluated JSONPath Expression com.jayway.jsonpath.JsonPath@5b8d184c but the path was not found; will use a null value
Successfully converted 2 records for FlowFile[1,627271185518617.mockFlowFile,23B]```

Reply:
I tried a couple writers, but at the very least the record delimiter is output, so it appears you still get a record even if it doesn't have any fields. This is good as it stands, thanks!